src/lib/factories/PathStartFactory.ts

```diff
@@ -0,0 +1,31 @@
+import path from 'node:path';
+import * as fg from 'fast-glob';
+import {getMessage, BundleName} from '../messages';
+
+export class PathStartFactory {
+	public async create(pathStarts?: string[]): Promise<string[]|undefined> {
+		if (pathStarts == null) {
+			return pathStarts;
+		}
+		const processedPaths: string[] = [];
+		for (const pathStart of pathStarts) {
+			// Globs assume the path separator character is '/' and treat '\\' as an escape character. So on OSs where
+			// the path separator character isn't '/', we need to convert paths into Glob syntax before checking whether
+			// they're actually Globs, to prevent misidentification.
+			const normalizedPathStart: string = path.sep === '/' ? pathStart : fg.convertPathToPattern(pathStart);
```

Comment:
Since you are leaving in the '#method;method2' syntax in there... does this impact the convertPathToPattern or isDynamicPattern methods?

Originally I thought we would have to do separate handline. My idea was to have:
```
if (pathStart.includes('#')) {
    const parts: string[] = pathStartPointStr.split('#');
    if (fg.isDynamicPattern(toPattern(parts[0]))) {
        throw new Error(getMessage(BundleName.PathStartFactory, 'error.glob-method-conflict', [pathStart]));
    }
    processedPaths.push(pathStart);
    continue;
}
```
where toPattern is:
```
function toPattern(value: string): string {
    return path.sep === '/' ? value : fg.convertPathToPattern(value);
}
```

Reply:
`#` isn't a character the globs care about, so we should be fine. See https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#what-is-a-static-or-dynamic-pattern